### DISCUSSION TITLE
Update of topic embeddings in update_topics

### DISCUSSION THREAD

**Discussion by slvnwhrl** [MaartenGr/BERTopic#2106] [General]
Hi there :)

I was wondering whether there is a specific reason that `_create_topic_vectors` in `update_topics` is called without passing the `docs` ([here](https://github.com/MaartenGr/BERTopic/blob/2353f4c21d74e33e34e30dbae938304bff094792/bertopic/_bertopic.py#L1574)) leading to a different calculation of the topic embeddings (i.e., not taking the average of all topic embeddings). Personally, I find this a bit unintuitive but I might be missing something. In the comments of the `update_topics` ([here](https://github.com/MaartenGr/BERTopic/blob/2353f4c21d74e33e34e30dbae938304bff094792/bertopic/_bertopic.py#L4100)) it says this is done because the documents are not known but they need to be passed to `update_topics`. So they could be passed on. 

Thanks!

**Comment by MaartenGr** [comment]
That's because you would also need the embeddings of the documents, not just the documents themselves to create the topic embeddings. As a result, it is much faster to create the topic embeddings based on the keyword representations rather than re-calculating all document embeddings. 

We could an additional `embeddings` parameter but that would then purely be for a potential recalculation of the topic embeddings (which also isn't always the case). 

Instead of the keyword-based approach, another technique would be to extract the top 10 documents per topic, embed and average those as the topic embedding.

**Reply by slvnwhrl** [reply]
Ok, that makes sense, especially the performance penalty. I got good results when comparing topics based on the average of all document embeddings, that's why I am asking. Comparing the document embeddings seems interesting to me, generally, especially when using an embedding model that was trained to compare text using cosine similarity.

Thanks for your swift answer!
